false-negative/unhandled-exception.cpp

```diff
@@ -0,0 +1,8 @@
+// Clang-Tidy cannot detect unhandled exceptions:
+// ones that are thrown and never caught (no corresponding catch block).
+
+#include <stdexcept>
+
+int main() {
+    throw std::logic_error("Unhandled exception (no catch block)!");
```

Comment:
@andrewt0301 what if we throw `int`? same problem?

Reply:
Throwing `int` will not be detected either. However, it is philosophical question whether throwing general types is acceptable in the first place (this is done when a programmer want to use exception as control-flow operation, which is a bad practice).

CLion says: `Clang-Tidy: Throwing an exception whose type 'int' is not derived from 'std::exception'.

However, `clang-tidy.exe` does not give an error in both cases.

I have added thowing `int` to the sake of completeness.